src/platform/Zephyr/ConfigurationManagerImpl.cpp

```diff
@@ -176,6 +176,7 @@ void ConfigurationManagerImpl::RunConfigUnitTest(void)
 void ConfigurationManagerImpl::DoFactoryReset(intptr_t arg)
 {
     ChipLogProgress(DeviceLayer, "Performing factory reset");
+    ThreadStackManager().LockThreadStack();
```

Comment:
This probably won't compile (?)
```suggestion
    ThreadStackMgr().LockThreadStack();
```

Also, do not assume it's Matter over **Thread**.

Reply:
@Damian-Nordic Is there any Matter api which allow to disable thread?

Reply:
Not sure I understand, I meant that you should compile this line only if `CONFIG_NET_L2_OPENTHREAD` is defined as it may not compile for Wi-Fi.

Reply:
Btw. you could add a comment explaining why this is needed as `Lock` without `Unlock` can look weird to newcomers :).